projects/packages/jetpack-mu-wpcom/src/features/wpcom-navigation-export-filter/class-export-filter.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * WPCOM Navigation Export Filter
+ *
+ * Filters wp_navigation posts from WXR exports when the post_author
+ * is not a valid user on the current site.
+ *
+ * @package automattic/jetpack-mu-wpcom
+ */
+
+declare( strict_types = 1 );
+
+namespace Automattic\Jetpack\Jetpack_Mu_Wpcom\Wpcom_Navigation_Export_Filter;
+
+/**
+ * Export Filter class to handle wp_navigation post filtering during WordPress exports.
+ */
+class Export_Filter {
+	/**
+	 * Tracks whether an export is currently in progress.
+	 *
+	 * @var bool
+	 */
+	private $is_exporting = false;
+
+	/**
+	 * Constructor to set up the export filter.
+	 */
+	public function __construct() {
+		add_action( 'export_wp', array( $this, 'start_export_filtering' ), 5 );
+	}
+
+	/**
+	 * Starts filtering export queries by hooking into the query filter.
+	 *
+	 * @return void
+	 */
+	public function start_export_filtering(): void {
+		$this->is_exporting = true;
+		add_filter( 'query', array( $this, 'filter_export_queries' ) );
+	}
+
+	/**
+	 * Stops filtering export queries by removing the query filter.
+	 *
+	 * @return void
+	 */
+	public function stop_export_filtering(): void {
+		$this->is_exporting = false;
+		remove_filter( 'query', array( $this, 'filter_export_queries' ) );
+	}
+
+	/**
+	 * Filters database queries during export to exclude wp_navigation posts
+	 * with invalid authors.
+	 *
+	 * @param string $query The database query.
+	 * @return string The modified query.
+	 */
+	public function filter_export_queries( string $query ): string {
+		global $wpdb;
+
+		if ( ! $this->is_exporting ) {
+			return $query;
+		}
+
+		// Target the specific query pattern from export_wp() that selects post IDs
+		$pattern = '/^SELECT ID FROM ' . preg_quote( $wpdb->posts, '/' ) . '.*WHERE/i';
```

Comment:
I don't think it should be a problem, but we're computing the pattern on every call. Moving it to the constructor would be more performant, but I don't think we're calling this method many times on a single thread, right?

Reply:
It's called a handful of times - but each time the string might change (in theory, not practice), as $wpdb->posts changes if `switch_to_blog` is used for example.

The users query should be fine too - the list of users should be cached by `get_users`